snuba/web/rpc/storage_routing/routing_strategies/outcomes_based.py

```diff
@@ -158,6 +158,13 @@ def _get_routing_decision(self, routing_context: RoutingContext) -> RoutingDecis
             can_run=True,
         )
         in_msg_meta = extract_message_meta(routing_decision.routing_context.in_msg)
+
+        thirty_days_ago_ts = int((datetime.now(tz=UTC) - timedelta(days=30)).timestamp())
+        older_than_thirty_days = thirty_days_ago_ts > in_msg_meta.start_timestamp.seconds
+
+        if state.get_int_config("enforce_unsampled_retention", 0) and older_than_thirty_days:
```

Comment:
I don't think the setting name very clear. Can we name it something like "enable_long_term_retention_downsampling" or something like this?